src/cloudant/database.py

```diff
@@ -299,3 +305,151 @@ def __iter__(self, remote=True):
                     yield doc
 
             raise StopIteration
+
+    def security_document(self):
+        """
+        _security_document_
+
+        Fetch the security document for this database
+        which contains information about who the database
+        is shared with
+
+        GET _api/v2/db/<dbname>/_security
+
+        """
+        resp = self._r_session.get(self.security_url)
+        resp.raise_for_status()
+        return resp.json()
+
+    def share_database(self, username, reader=True, writer=False, admin=False):
+        """
+        _share_database_
+
+        Share this database with the username provided.
+        You can grant varying degrees of access rights,
+        default is to share read-only, but writing or admin
+        permissions can be added by setting the appropriate flags
+        If the user already has this database shared with them it
+        will modify/overwrite the existing permissions
+
+        :param username: Cloudant Username to share the database with
+        :param reader: Grant named user read access if true
+        :param writer: Grant named user write access if true
+        :param admin: Grant named user admin access if true
+
+        """
+        doc = self.security_document()
+        data = doc.get('cloudant', {})
+        perms = []
+        if reader:
+            perms.append('_reader')
+        if writer:
+            perms.append('_writer')
+        if admin:
+            perms.append('_admin')
+
+        data[username] = perms
+        resp = self._r_session.put(
+            self.security_url,
+            data=json.dumps(doc),
+            headers={'Content-Type': 'application/json'}
+            )
+        resp.raise_for_status()
+        return resp.json()
+
+    def unshare_database(self, username):
+        """
+        _unshare_database_
+
+        Remove all sharing with the named user for this database.
+        This will remove the entry for the user from the security doc
+        To modify permissions, instead of remove thame,
+        use the share_database method
+
+        """
+        doc = self.security_document()
+        data = doc.get('cloudant', {})
+        if username in data:
+            del data[username]
+        resp = self._r_session.put(
+            self.security_url,
+            data=json.dumps(doc),
```

Comment:
Same as the above...sort of...because if it's empty this is basically a noop. In addition, rather than using security url we actually already expose this via DELETE _api/v2/user/shared_databases.